packages/lesswrong/components/search/SearchPageTabbed.tsx

```diff
@@ -0,0 +1,326 @@
+    // are doing a somewhat hacky thing to style it.
+    "& .ais-SearchBox-input": {
+      height: "100%",
+      width: "100%",
+      paddingRight: 0,
+      verticalAlign: "bottom",
+      borderStyle: "none",
+      boxShadow: "none",
+      backgroundColor: "transparent",
+      fontSize: 'inherit',
+      "-webkit-appearance": "none",
+      cursor: "text",
+      ...theme.typography.body2,
+    },
+  },
+  tabs: {
+    margin: '0 auto 20px',
+    '& .MuiTab-root': {
+      minWidth: 110,
+      [theme.breakpoints.down('xs')]: {
+        minWidth: 50
+      }
+    },
+    '& .MuiTab-labelContainer': {
+      fontSize: '1rem'
+    }
+  },
+  resultCount: {
+    fontFamily: theme.typography.fontFamily,
+    fontWeight: 400,
+    fontSize: 14,
+    color: theme.palette.grey[700],
+    marginBottom: 20
+  },
+  
+  pagination: {
+    ...theme.typography.commentStyle,
+    fontSize: 16,
+    '& li': {
+      padding: 8
+    },
+    '& .ais-Pagination-item': {
+      color: theme.palette.primary.main,
+    },
+    '& .ais-Pagination-item--firstPage': {
+      paddingLeft: 0
+    },
+    '& .ais-Pagination-item--page': {
+      fontWeight: 600
+    },
+    '& .ais-Pagination-item--selected': {
+      color: theme.palette.grey[900]
+    },
+    '& .ais-Pagination-item--disabled': {
+      color: theme.palette.grey[500]
+    }
+  }
+})
+
+// filters by tags
+const TagsRefinementList = ({ refine, tagsFilter, setTagsFilter }) => {
+  return <Components.TagMultiselect
+    value={tagsFilter}
+    path="tags"
+    placeholder={`Filter by ${taggingNamePluralSetting.get()}`}
+    hidePostCount
+    updateCurrentValues={(values: {tags?: Array<string>}) => {
+      setTagsFilter(values.tags)
+      refine(values.tags)
+    }}
+  />
+}
+const CustomTagsRefinementList = connectRefinementList(TagsRefinementList)
+
+// shows total # of results
+const ResultsCount = ({ searchResults, className }) => {
+  if (!searchResults) return null
+  
+  return <div className={className}>
+    {searchResults.nbHits} result{searchResults.nbHits === 1 ? '' : 's'}
+  </div>
+}
+const CustomTotalStateResults = connectStateResults(ResultsCount)
+
+
+const SearchPageTabbed = ({classes}:{
+  classes: ClassesType
+}) => {
+  const { history } = useNavigation()
+  const { location, query } = useLocation()
+  
+  // store these values for the search filter
+  const pastDay = useRef(moment().subtract(24, 'hours').valueOf())
+  const pastWeek = useRef(moment().subtract(7, 'days').valueOf())
+  const pastMonth = useRef(moment().subtract(1, 'months').valueOf())
+  const pastYear = useRef(moment().subtract(1, 'years').valueOf())
+
+  // initialize the tab & search filters from the URL
+  const [tab, setTab] = useState<AlgoliaIndexCollectionName>(() => {
+    const contentType = query.contentType as AlgoliaIndexCollectionName
+    return collectionIsAlgoliaIndexed(contentType) ? contentType : 'Posts'
+  })
```

Comment:
Not sure if I did this right (it removed the error at least) - let me know if there is a more TS way

Reply:
`collectionIsAlgoliaIndexed` expects a `CollectionNameString`, but functionally it won't care if you pass in any arbitrary string, since it's checking whether an (narrower) array of strings contains the provided string.  If I was doing it from scratch I might do something like this:
```
const algoliaCollectionNames = {
  Comments: 1,
  Posts: 1,
  Users: 1,
  Sequences: 1,
  Tags: 1
};

const isAlgoliaCollectionName = (name: string): name is keyof typeof algoliaCollectionNames => {
  return name in algoliaCollectionNames;
};
```

But that'd involve refactoring the existing logic already using `algoliaIndexedCollectionNames`, which doesn't seem worth it.  So this is probably the best available option unless you want to refactor a bunch of other stuff.